tests/test_slim_engine.py

```diff
@@ -1020,12 +1020,31 @@ def verify_genomic_elements(self, contig, ts):
         ge_types = self.slim_metadata_key0(
             ts.metadata["SLiM"]["user_metadata"], "genomicElementTypes"
         )
+        # getting the ids for the neutral mutation types
+        neutral_ids = [
+            mtdict["id"]
+            for mtdict in contig.mutation_types()
+            if mtdict["mutation_type"].is_neutral
+        ]
+
         assert len(contig.dfe_list) == len(ge_types)
         for j, (dfe, intervals) in enumerate(
             zip(contig.dfe_list, contig.interval_list)
         ):
             assert str(j) in ge_types
             ge = self.slim_metadata_key0(ge_types, str(j))
+            # figuring out if all mutation types are neutral in this GET
+            all_neutral = all(
+                [mtype_id in neutral_ids for mtype_id in ge["mutationTypes"]]
+            )
+            # checking that the neutral mutations have 0.0 proportion unless
+            # all the mutations are neutral in this dfe
+            for mtype_id, prop in zip(ge["mutationTypes"], ge["mutationFractions"]):
+                if mtype_id in neutral_ids:
+                    if all_neutral:
+                        assert prop > 0.0
+                    else:
+                        assert np.isclose(prop, 0.0)
```

Comment:
One might have an "all neutral" DFE with a proportion that is 0.0, so this isn't quite right. How about:
```suggestion
            for mt, prop in zip(dfe.mutation_types, ge["mutationFractions"]):
                if mt.is_neutral or all_neutral:
                         assert np.isclose(prop, 0.0)
```

Reply:
No, that would trigger an error in SLiM actually (you cannot have a genomic element type with all the proportions equal to 0). So I should add a check to the `slim_engine` to catch that.

Reply:
That's not what I mean - I mean an all neutral DFE where some of the mutation types have nonzero proportion and others have a zero proportion. (e.g., the proportions are `[1.0, 0.0]`).